c_binding/lsm_local_disk.c

```diff
@@ -804,3 +795,75 @@ static int _sysfs_sas_addr_get(const char *blk_name, char *tp_sas_addr)
 
     return rc;
 }
+
+static int _sas_addr_get(char *err_msg, const char *disk_path,
+                         char *tp_sas_addr)
+{
+    int rc = LSM_ERR_OK;
+    int fd = -1;
+
+    assert(disk_path != NULL);
+    assert(tp_sas_addr != NULL);
+
+    memset(tp_sas_addr, 0, _SG_T10_SPL_SAS_ADDR_LEN);
+
+    /* TODO(Gris Ge): Add support of NVMe enclosure */
+
+    /* Try use sysfs first to get SAS address. */
+    if ((strlen(disk_path) > strlen("/dev/")) &&
+        (strncmp(disk_path, "/dev/", strlen("/dev/")) == 0) &&
+        (strncmp(disk_path + strlen("/dev/"), "sd", strlen("sd")) == 0))
+        _sysfs_sas_addr_get(disk_path + strlen("/dev/"), tp_sas_addr);
```

Comment:
_sysfs_sas_addr_get does have a return value that you are ignoring.  Wouldn't it be better to use the same rc == LSM_ERR_OK internally and then wrap this call with `_good` too?